CRUD/AddCart/Observer/Product/Data.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace CRUD\AddCart\Observer\Product;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use CRUD\AddCart\Model\AddcartFactory as CartModel;
+use CRUD\AddCart\Model\ResourceModel\Addcart as CartResource;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Http\Context;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use CRUD\AddCart\Publisher\Addcart;
+
+class Data implements ObserverInterface
+{
+
+    /**
+     * @var Context
+     */
+    protected $httpContext;
+
+    /**
+     * @var CartModel
+     */
+    protected $model;
+    /**
+     * @var Addcart
+     */
+    protected $publisher;
+
+    /**
+     * @var CartResource
+     *
+     */
+
+    protected $resource;
+
+    /**
+     * @param CartModel $model
+     * @param CartResource $resource
+     * @param Context $httpContext
+     * @param Addcart $publisher
+     */
+    protected $_logger;
+    private $checkoutSession;
+
+    public function __construct(
+        \Psr\Log\LoggerInterface $_logger,
```

Comment:
@msakila532 Don't mix and match the fully qualified class names here. Either use alias here or declare fully for all classes